src/test/java/io/fusionauth/http/MultipartTest.java

```diff
@@ -86,16 +90,177 @@ public void post(String scheme) throws Exception {
     };
 
     CountingInstrumenter instrumenter = new CountingInstrumenter();
-    try (HTTPServer ignore = makeServer(scheme, handler, instrumenter).start()) {
+    try (HTTPServer ignore = makeServer(scheme, handler, instrumenter)
+        .withMultipartConfiguration(new MultipartConfiguration().withFileUploadEnabled(true))
+        .start()) {
       URI uri = makeURI(scheme, "");
       var client = makeClient(scheme, null);
       var response = client.send(
-          HttpRequest.newBuilder().uri(uri).header(Headers.ContentType, "multipart/form-data; boundary=----WebKitFormBoundaryTWfMVJErBoLURJIe").POST(BodyPublishers.ofString(Body)).build(),
+          HttpRequest.newBuilder()
+                     .uri(uri)
+                     .header(Headers.ContentType, "multipart/form-data; boundary=----WebKitFormBoundaryTWfMVJErBoLURJIe")
+                     .POST(BodyPublishers.ofString(Body)).build(),
           r -> BodySubscribers.ofString(StandardCharsets.UTF_8)
       );
 
       assertEquals(response.statusCode(), 200);
       assertEquals(response.body(), ExpectedResponse);
     }
   }
-}
\ No newline at end of file
+
+  @Test(dataProvider = "schemes")
+  public void post_server_configuration_fileTooBig(String scheme) throws Exception {
+    // File too big even though the overall request size is ok.
+    withScheme(scheme)
+        .withFileSize(513) // 513 bytes
+        .withConfiguration(new MultipartConfiguration().withFileUploadEnabled(true)
+                                                       // Max file size is 2Mb
+                                                       .withMaxFileSize(512)
+                                                       // Max request size is 10 Mb
+                                                       .withMaxRequestSize(10 * 1024 * 1024))
+        .expect(response -> assertEquals(response.statusCode(), 413));
+  }
+
+  @Test(dataProvider = "schemes")
+  public void post_server_configuration_file_upload_disabled(String scheme) throws Exception {
+    // File uploads disabled
+    withScheme(scheme)
```

Comment:
Is this correct? If files are disabled, should we be returning an error or should we just ignore them and still handle the rest of the request? I can go either way, but seems like you were planning on looking at the action in Prime and disabling files if the action didn't need them, which would blow up if the request has files, which we could just ignore.